src/bin/pg_autoctl/keeper.c

```diff
@@ -947,6 +946,28 @@ keeper_ensure_configuration(Keeper *keeper, bool postgresNotRunningIsOk)
 			}
 		}
 	}
+	else
```

Comment:
We have many states where we don't want to ALTER ROLE PASSSWORD, such as MAINTENANCE and its related states. I think this should not be an else, but rather its own `if`. I think the list of states in which we want to act on the password change are SINGLE, WAIT_PRIMARY, JOIN_PRIMARY, PRIMARY, APPLY_SETTINGS.